### ISSUE TITLE
LogicConditionNeedOptimization: false positive when many conditions are wrapped in parenthesis

### ISSUE THREAD

**Issue by rnveach** [sevntu-checkstyle/sevntu.checkstyle#683]
````
$ cat TestClass.java
import java.nio.*;

public class NativeOrder {
    public static void main(String[] args) throws Exception {
        ByteOrder bo = ByteOrder.nativeOrder();
        System.err.println(bo);
        String arch = System.getProperty("os.arch");
        if (((arch.equals("sparc") && (bo != ByteOrder.BIG_ENDIAN))) ||
            ((arch.equals("i386") && (bo != ByteOrder.LITTLE_ENDIAN)))) {
            throw new Exception("Wrong byte order");
        }
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="com.github.sevntu.checkstyle.checks.coding.LogicConditionNeedOptimizationCheck">
</module>
    </module>
</module>

$ java -jar checkstyle-8.9-sevntu-1.29.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:8: Condition with && at line 8 position 35 need optimization. All method calls are advised to move to end of logic expression. [LogicConditionNeedOptimization]
[ERROR] TestClass.java:8: Condition with || at line 8 position 69 need optimization. All method calls are advised to move to end of logic expression. [LogicConditionNeedOptimization]
[ERROR] TestClass.java:9: Condition with && at line 9 position 34 need optimization. All method calls are advised to move to end of logic expression. [LogicConditionNeedOptimization]
Audit done.
Checkstyle ends with 3 errors.
````

Violations for 8:35 and 9:34 are valid.
The violation at 8:69 is not. The `||` between the 2 big blocks both have method calls, so there is no better ordering switching them around, and the violation will stay even if they are switched.
Once the other 2 violations are fixed, the other one disappears.

I couldn't create an example that only involved 1 violation.

**Comment by rnveach** [comment]
The issue is LPAREN and RPAREN aren't nested in each other in this type of situation.

````
    |      |  |  `--LOR -> || [8:69]
    |      |  |      |--LPAREN -> ( [8:12]
    |      |  |      |--LPAREN -> ( [8:13]
    |      |  |      |--LAND -> && [8:35]
....
    |      |  |      |--RPAREN -> ) [8:66]
    |      |  |      |--RPAREN -> ) [8:67]
    |      |  |      |--LPAREN -> ( [9:12]
    |      |  |      |--LPAREN -> ( [9:13]
    |      |  |      |--LAND -> && [9:34]
````

So instead of comparing `(a && b)` to (c && d)`, are probably comparing `(a && b)` to `d`.
So instead of doing a single `branchContains`, we would have to do it on all inside the first set of parenthesis.

**Comment by rnveach** [comment]
Fix was merged
